src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFSDriver.Linux.cs

```diff
@@ -0,0 +1,562 @@
+                {
+                    // Wrapping the exception in order to get a better message.
+                    throw new UnauthorizedAccessException("Opening pins requires root permissions.", e);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Closes an open pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        protected internal override void ClosePin(int pinNumber)
+        {
+            string pinPath = $"{GpioBasePath}/gpio{pinNumber}";
+            // If the directory doesn't exist, this becomes a no-op since the pin was closed already.
+            if (Directory.Exists(pinPath))
+            {
+                try
+                {
+                    SetPinEventsToDetect(pinNumber, PinEventTypes.None);
+                    File.WriteAllText(Path.Combine(GpioBasePath, "unexport"), pinNumber.ToString());
+                    _exportedPins.Remove(pinNumber);
+                }
+                catch (UnauthorizedAccessException e)
+                {
+                    throw new UnauthorizedAccessException("Closing pins requires root permissions.", e);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the mode to a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+            {
+                throw new PlatformNotSupportedException("This driver is generic so it does not support Input Pull Down or Input Pull Up modes.");
+            }
+            string directionPath = $"{GpioBasePath}/gpio{pinNumber}/direction";
+            string sysfsMode = ConvertPinModeToSysFsMode(mode);
+            if (File.Exists(directionPath))
+            {
+                try
+                {
+                    File.WriteAllText(directionPath, sysfsMode);
+                }
+                catch (UnauthorizedAccessException e)
+                {
+                    throw new UnauthorizedAccessException("Setting a mode to a pin requires root permissions.", e);
+                }
+            }
+            else
+            {
+                throw new InvalidOperationException("There was an attempt to set a mode to a pin that is not open.");
+            }
+        }
+
+        private string ConvertPinModeToSysFsMode(PinMode mode)
+        {
+            if (mode == PinMode.Input)
+            {
+                return "in";
+            }
+            if (mode == PinMode.Output)
+            {
+                return "out";
+            }
+            throw new PlatformNotSupportedException($"{mode} is not supported by this driver.");
```

Comment:
Is this necessary? Just use `Debug.Assert` since it looks like we already validate that mode is either `PinMode.Input` or `PinMode.Output` in the caller.